standard_sequence/classy_sequence.py

```diff
@@ -861,263 +318,528 @@ def _do_molasses_tof_sequence(self, t, reset_mot = False):
+        return t
 
     def kill_F3(self, t):
         pass
 
+    def _optical_pump_molasses_sequence(self, t, reset_mot=False):
+        # MOT loading time 500 ms
+        mot_load_dur = 0.5
 
+        t += D2Lasers.CONST_SHUTTER_TURN_ON_TIME
 
-class TweezerSequence(MOTSequence):
+        t = self.do_mot(t, mot_load_dur)
+        t = self.do_molasses(t, shot_globals.bm_time)
+
+        t = self.pump_to_F4(t)
+
+        t = self.do_molasses_dipole_trap_imaging(t, close_all_shutters=True)
+
+        # Turn off MOT for taking background images
+        t += 1e-1
+
+        t = self.do_molasses_dipole_trap_imaging(t, close_all_shutters=True)
+        t += 1e-2
+
+        if reset_mot:
+            t = self.reset_mot(t)
+
+        return t
+
+    def _do_pump_debug_in_molasses(self, t, reset_mot=False):
+        mot_load_dur = 0.5
+        t = self.do_mot(t, mot_load_dur)
+        t = self.do_molasses(t, shot_globals.bm_time)
+
+
+        if shot_globals.do_dp:
+            t, t_aom_off= self.depump_to_F3(t, shot_globals.op_label)
+        if shot_globals.do_op:
+            t, t_aom_off = self.pump_to_F4(t, shot_globals.op_label, close_all_shutters = False)
+        if shot_globals.do_depump_pulse_after_pumping:
+            t = self.depump_ta_pulse(t_aom_off, close_all_shutters = True)
+        if shot_globals.do_killing_pulse:
+            t = self.kill_F4(t_aom_off, shutter_config = ShutterConfig.OPTICAL_PUMPING_FULL)
+        t_depump = t
+
+        t = self.BField_obj.ramp_bias_field(
+            t,
+            bias_field_vector=(
+                shot_globals.mw_biasx_field,
+                shot_globals.mw_biasy_field,
+                shot_globals.mw_biasz_field,
+            ),
+        )
+
+        if shot_globals.do_mw_pulse:
+            t = self.Microwave_obj.do_pulse(t, shot_globals.mw_time)
+
+        if shot_globals.do_killing_pulse:
+            t = self.kill_F4(t_aom_off, shutter_config = ShutterConfig.OPTICAL_PUMPING_FULL)
+
+        # postpone next sequence until shutter off time reached
+        t = max(t, t_depump + D2Lasers.CONST_MIN_SHUTTER_OFF_TIME)
+
+        # This is the only place required for the special value of imaging
+        t = self.do_molasses_dipole_trap_imaging(
+            t,
+            ta_power=0.1,
+            repump_power=1,
+            exposure=10e-3,
+            do_repump=shot_globals.mw_imaging_do_repump,
+            close_all_shutters=True,
+        )
+
+        # Turn off MOT for taking background images
+        t += 1e-1
+
+        t = self.do_molasses_dipole_trap_imaging(
+            t,
+            ta_power=0.1,
+            repump_power=1,
+            exposure=10e-3,
+            do_repump=shot_globals.mw_imaging_do_repump,
+            close_all_shutters=True,
+        )
+        t += 1e-2
+        t = self.Microwave_obj.reset_spectrum(t)
+        if reset_mot:
+            t = self.reset_mot(t)
+
+        return t
+
+    def _do_F4_microwave_spec_molasses(self, t, reset_mot=False):
```

Comment:
Add a docstring explaining what this sequence is for. There are several custom function calls here. Should this be broken up any further into logical sub-blocks?

Reply:
added